_posts/2020-11-04-solidity-ama-1-recap.md

```diff
@@ -35,35 +35,80 @@ You can check the progress [here](https://github.com/ethereum/solidity/pull/1001
 
 The [table of builtins](https://solidity.readthedocs.io/en/develop/yul.html#evm-dialect) should provide a good summary. If you need more details about the EVM, maybe try the [section on the EVM](https://solidity.readthedocs.io/en/develop/introduction-to-smart-contracts.html#index-6). If you are missing something, please feel free to open an issue in the [Solidity repo](https://github.com/ethereum/solidity/issues)!
 
-### Solidity Optimizer
+#### Rollups: What do you think of optimism's solc customisations? Is this a viable way to implement containerisation?
+
+I hope they will switch to a yul-based approach once the code generator is finished. The code generated through the IR does not contain any dynamic jumps and it can be easily rewritten by a very simple tool to do what optimism needs.
 
-Clarify difference between Yul and Solidity Optimizer here?
+### Solidity Optimizer
 
 #### What is the Solidity Optimizer optimizing for (size or cost or something else?) and how does it achieve that?
 
 The Solidity compiler uses two different optimizer modules: The "old" optimizer that operates at opcode level and the "new" optimizer that operates on Yul IR code. The opcode-based optimizer applies simplification rules from the [list](https://github.com/ethereum/solidity/blob/develop/libevmasm/RuleList.h) to opcodes next to each other. It also combines equal code sets, removes unused code and some other things. The Yul-based optimizer is much more powerful, because it can work across function calls: In Yul it is not possible to perform arbitrary jumps, so it is for example possible to compute the side-effects of each function: If a function does not modify storage, a call to it can be swapped with a function that does. If a function is side-effect free and its result is multiplied by zero, you can remove the function call completely.
 
+There is single stage, the "constant optimizer" where the trade-off between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant. The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.
```

Comment:
Regarding https://github.com/ethereum/solidity-blog/pull/106#discussion_r517212279
> For me this paragraph is quite hard to read and did not make sense as an introduction. Do we need it or is it ok to skip it? If we need it, we should try to reword it. Especially the "there is single stage" does not seem to make sense if you don't introduce / compare it to another stage. Which optimizer is this part about? (Or maybe I'm missing that, but it's definitely not a good introduction. Mabye we can move it as the second paragraph?)

How about this:

```suggestion
In the opcode-based optimizer there is only one stage (the "constant optimizer") where the trade-off between deploy-time and run-time costs are taken into account. This stage tries to find a "better" representation of each number in the source, like ``0x10000000000`` can be encoded as ``PUSH6 0x10000000000`` (7 bytes and almost zero run-time costs), but it can also be encoded as `PUSH1 1 PUSH1 40 SHL` (5 bytes and a bit more expensive at run-time). Most of the time, the difference is not too relevant.

The optimizer tries to simplify complicated expressions (which reduces both size and execution cost), but it also specializes or inlines functions. Especially function inlining is an operation that can cause much bigger code, but it is often done because it results in opportunities for more simplifications.
```

I split the paragraph because I think the first part is only relevant for the opcode-based optimizer while the second one only for yul-optimizer (the old one does not do inlining).